quill-engine/src/main/scala/io/getquill/idiom/StatementInterpolator.scala

```diff
@@ -6,31 +6,34 @@ import io.getquill.util.Messages._
 
 import scala.collection.mutable.ListBuffer
 
+//noinspection ConvertExpressionToSAM
 object StatementInterpolator {
 
   trait Tokenizer[T] {
     def token(v: T): Token
   }
 
   object Tokenizer {
-    def apply[T](f: T => Token) = new Tokenizer[T] {
-      def token(v: T) = f(v)
-    }
+    def apply[T](f: T => Token): Tokenizer[T] =
+      new Tokenizer[T] {
+        def token(v: T): Token = f(v)
+      }
+
     def withFallback[T](
       fallback: Tokenizer[T] => Tokenizer[T]
-    )(pf: PartialFunction[T, Token]) =
+    )(pf: PartialFunction[T, Token]): Tokenizer[T] =
       new Tokenizer[T] {
-        private val stable       = fallback(this)
-        override def token(v: T) = pf.applyOrElse(v, stable.token)
+        private lazy val stable: Tokenizer[T] = fallback(this)
```

Comment:
This was weird to me. Why would we need to compute the fallback value before to know if we'll ever need it one day? 🤔